8627 perceptual loss errors out after hitting the maximum number of downloads

virginiafdez opened the pull request:
Fixes #8627.
Moves the perceptual loss code to MONAI repository https://github.com/Project-MONAI/perceptual-models and the checkpoints to Huggingface. 

### Description
This PR changes and simplifies the torch.hub loading process and gets the models from Huggingface lirbary. 
A few sentences describing the changes proposed in this pull request.

### Types of changes
<!--- Put an `x` in all the boxes that apply, and remove the not applicable items -->
- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Breaking change (fix or new feature that would cause existing functionality to change).
- [N/A] New tests added to cover the changes.
- [N/A] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x]  In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

Commit:
Perceptual loss changes.

Commit:
Merge branch '8627-perceptual-loss-errors-out-after-hitting-the-maximum-number-of-downloads' of github.com:virginiafdez/MONAI into 8627-perceptual-loss-errors-out-after-hitting-the-maximum-number-of-downloads

Commit:
Unnecessary import

Commit:
Fixes

virginiafdez committed:
Merge branch 'dev' into 8627-perceptual-loss-errors-out-after-hitting-the-maximum-number-of-downloads

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

ericspod commented on monai/losses/perceptual.py:
```diff
@@ -205,11 +211,17 @@ class MedicalNetPerceptualSimilarity(nn.Module):
     def __init__(
-        self, net: str = "medicalnet_resnet10_23datasets", verbose: bool = False, channel_wise: bool = False
+        self,
+        net: str = "medicalnet_resnet_10_23datasets",
```
What we should do to improve security is to move "medicalnet_resnet10_23datasets" and any other valid name into constants then check `net` is one of them.

virginiafdez replied:
Done.

ericspod commented on monai/losses/perceptual.py:
```diff
@@ -205,11 +211,17 @@ class MedicalNetPerceptualSimilarity(nn.Module):
         super().__init__()
         torch.hub._validate_not_a_forked_repo = lambda a, b, c: True
-        self.model = torch.hub.load("warvito/MedicalNet-models", model=net, verbose=verbose, trust_repo=True)
+        self.model = torch.hub.load(
```
Perhaps this should have `trust_repo=True` as well since we do trust this repo and it's name is hardcoded here.

virginiafdez replied:
Done.

ericspod commented on monai/losses/perceptual.py:
```diff
@@ -205,11 +211,17 @@ class MedicalNetPerceptualSimilarity(nn.Module):
     """
 
     def __init__(
```
We might need to check that `huggingface_hub` is installed as well, the `torch.hub.load` call will fail if so.

virginiafdez replied:
Done.

Commit:
Merge branch '8627-perceptual-loss-errors-out-after-hitting-the-maximum-number-of-downloads' of github.com:virginiafdez/MONAI into 8627-perceptual-loss-errors-out-after-hitting-the-maximum-number-of-downloads

Commit:
Add check of network name

ericspod committed:
Update monai/losses/perceptual.py

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>
Signed-off-by: Eric Kerfoot <17726042+ericspod@users.noreply.github.com>

ericspod committed:
Update monai/losses/perceptual.py

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>
Signed-off-by: Eric Kerfoot <17726042+ericspod@users.noreply.github.com>

ericspod committed:
Update monai/losses/perceptual.py

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>
Signed-off-by: Eric Kerfoot <17726042+ericspod@users.noreply.github.com>

virginiafdez committed:
Update monai/losses/perceptual.py

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>
Signed-off-by: Virginia Fernandez <61539159+virginiafdez@users.noreply.github.com>

Commit:
Bug

Commit:
Merge branch '8627-perceptual-loss-errors-out-after-hitting-the-maximum-number-of-downloads' of github.com:virginiafdez/MONAI into 8627-perceptual-loss-errors-out-after-hitting-the-maximum-number-of-downloads

Commit:
DCO Remediation Commit for Virginia Fernandez <virginia.fernandez@kcl.ac.uk>

I, Virginia Fernandez <virginia.fernandez@kcl.ac.uk>, hereby add my Signed-off-by to this commit: b1e4a50d437ea96499aac709eaf50ce3f0e84232
I, Virginia Fernandez <virginia.fernandez@kcl.ac.uk>, hereby add my Signed-off-by to this commit: fa0639be83e9ebd97ed494174a7891500fadb067
I, Virginia Fernandez <virginia.fernandez@kcl.ac.uk>, hereby add my Signed-off-by to this commit: 5594bfe47f217134c35d29ca4e93f15f20ef7fe7
I, Virginia Fernandez <virginia.fernandez@kcl.ac.uk>, hereby add my Signed-off-by to this commit: c99e16eb050848681d83e211205e8eaa998606c4
I, Virginia Fernandez <virginia.fernandez@kcl.ac.uk>, hereby add my Signed-off-by to this commit: e3be8de4cd7c5b643d856d676875f755c08c84d4

Signed-off-by: Virginia Fernandez <virginia.fernandez@kcl.ac.uk>

Commit:
Reformatting

Signed-off-by: Virginia Fernandez <virginia.fernandez@kcl.ac.uk>

KumoLiu committed:
fix no space left issue

Signed-off-by: Yun Liu <yunl@nvidia.com>

KumoLiu committed:
fix typo in error message

Signed-off-by: Yun Liu <yunl@nvidia.com>

KumoLiu committed:
try fix no space left in packaging pipeline

Signed-off-by: Yun Liu <yunl@nvidia.com>

KumoLiu commented on .github/workflows/pythonapp.yml:
- Removed the cache weekly timestamp step from the packaging job.
- Removed the heavyweight dependency installs that were triggering OSError: [Errno 28] No space left on device:
- Stopped installing the source archive with "[all]" extras (avoids large optional deps like itk, mlflow, clearml, lpips, etc.).
- Stopped installing requirements-dev.txt in the packaging quick-test phase; switched to requirements-min.txt + tests/min_tests.py.
- Switched packaging to CPU-only PyTorch on Ubuntu using the PyTorch CPU index, and installed it with --no-cache-dir to avoid caching huge wheels.
- Reintroduced a controlled pip cache for packaging:
     Cache path is pip cache dir.
     Cache key uses hashFiles('requirements*.txt', 'setup.cfg', 'pyproject.toml') so it updates when dependencies change.
      Kept torch installs --no-cache-dir to reduce disk pressure, while allowing smaller deps (e.g., requirements-min.txt) to benefit from caching.

cc @ericspod

ericspod replied:
This file has been renamed and revamped with many of these changes integrated. It works now to resolve the issues with running out of space so we should delete this file to allow merging.

KumoLiu committed:
fix setup

Signed-off-by: Yun Liu <yunl@nvidia.com>

KumoLiu commented:
/build

KumoLiu commented:
/build

garciadias committed:
Merge branch 'dev' into 8627-perceptual-loss-errors-out-after-hitting-the-maximum-number-of-downloads

Signed-off-by: Rafael Garcia-Dias <rafaelagd@gmail.com>

ericspod committed:
Merge branch 'dev' into 8627-perceptual-loss-errors-out-after-hitting-the-maximum-number-of-downloads

ericspod committed:
Merge branch 'dev' into 8627-perceptual-loss-errors-out-after-hitting-the-maximum-number-of-downloads

ericspod committed:
Removing old file

Signed-off-by: Eric Kerfoot <17726042+ericspod@users.noreply.github.com>

ericspod committed:
Merge branch 'dev' into 8627-perceptual-loss-errors-out-after-hitting-the-maximum-number-of-downloads

ericspod commented:
Hi @virginiafdez we should be able to get this through now, I removed a conflicting file but otherwise there's a few minor things to address. Would you be able to get back to this?

virginiafdez committed:
Merge branch 'dev' into 8627-perceptual-loss-errors-out-after-hitting-the-maximum-number-of-downloads

Commit:
Fix lower() bug

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Commit:
stack_level and suggested tests

Commit:
Merge branch '8627-perceptual-loss-errors-out-after-hitting-the-maximum-number-of-downloads' of github.com:virginiafdez/MONAI into 8627-perceptual-loss-errors-out-after-hitting-the-maximum-number-of-downloads

virginiafdez committed:
Merge branch 'dev' into 8627-perceptual-loss-errors-out-after-hitting-the-maximum-number-of-downloads

Commit:
remove | operator on Enum

Commit:
Merge branch '8627-perceptual-loss-errors-out-after-hitting-the-maximum-number-of-downloads' of github.com:virginiafdez/MONAI into 8627-perceptual-loss-errors-out-after-hitting-the-maximum-number-of-downloads

Commit:
autofix run

Signed-off-by: Virginia Fernandez <virginia.fernandez@kcl.ac.uk>

Commit:
remove stacklevel

Commit:
ruff

Signed-off-by: Virginia Fernandez <virginia.fernandez@kcl.ac.uk>

ericspod commented:
Thanks! I'll set the DCO to pass as you said that was fine to do as it's all your code.
